chore(docs): return replace docs

Mkassabov opened the pull request:
At some point we lost these docs.

Tying this to [PR#619](https://github.com/sam-goodwin/alchemy/pull/619) since we make replace require an await call

sam-goodwin commented:
Oh wait, I removed the replace docs and moved them to resource.md (along with adopt) to keep the doc tree a bit more focused. https://alchemy.run/concepts/resource/#replacement

Mkassabov commented:
> Oh wait, I removed the replace docs and moved them to resource.md (along with adopt) to keep the doc tree a bit more focused. https://alchemy.run/concepts/resource/#replacement

<img width="2535" height="1578" alt="image" src="https://github.com/user-attachments/assets/cc9bbf4e-88f0-40bf-9233-fa610672e96b" />


Is there a reason its not on the sidebar on the right? I swear I checked this but I guess I missed it cause I didn't scroll all the way down?

sam-goodwin commented:
> Is there a reason its not on the sidebar on the right? I swear I checked this but I guess I missed it cause I didn't scroll all the way down?

Probably using a ### instead of a ## ? We should fix it

sam-goodwin commented:
@claude pleae make sure replace and adopt show up on side bar by using ## instead of ###
